src/contexts/Extensions/index.tsx

```diff
@@ -52,17 +51,26 @@ export const ExtensionsProvider = ({
   };
 
   const getInstalledExtensions = () => {
-    const { injectedWeb3 }: AnyApi = window;
-    const _exts: Extension[] = [];
-    EXTENSIONS.forEach((e: ExtensionConfig) => {
-      if (injectedWeb3[e.id] !== undefined) {
-        _exts.push({
-          ...e,
-          ...injectedWeb3[e.id],
-        });
-      }
-    });
-    return _exts;
+    if (!('injectedWeb3' in window && window.injectedWeb3)) return [];
+
+    const { injectedWeb3 } = window;
+    if (!isRecord(injectedWeb3)) return [];
+
+    return EXTENSIONS.reduce<Extension[]>((acc, extension) => {
```

Comment:
nit: Imho flatMap or filter -> map would make it more readable

Reply:
[Done](https://github.com/Cardinal-Cryptography/aleph-zero-dashboard/compare/cc130aa89fb6fad8ad5d0d1f70f93772f5dee007..205d7c8fc566c8ab64dee889751fcc405347c7f4)